plugin/src/main/java/io/fabric8/maven/plugin/BuildMojo.java

```diff
@@ -241,7 +293,191 @@ private void executeOpenShiftBuild(ServiceHub hub, ImageConfiguration imageConfi
+            if (updated) {
+                // lets store the entities again!
+                KubernetesList entity = new KubernetesListBuilder().withItems(new ArrayList<>(entities)).build();
+                File resourceFileBase = new File(this.targetDir, ResourceClassifier.OPENSHIFT.getValue());
+                AbstractResourceMojo.writeResourcesIndividualAndComposite(entity, resourceFileBase, ResourceFileType.yaml, log);
+            }
+        } catch (KubernetesClientException e) {
+            KubernetesResourceUtil.handleKubernetesClientException(e, this.log);
+        } catch (MojoExecutionException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+    }
+
+    private boolean updateImageStreamTag(OpenShiftClient client, ImageConfiguration imageConfig, ImageStream is, String buildConfigName) throws MojoExecutionException {
+        String namespace = client.getNamespace();
+        String imageName = imageConfig.getName();
+        String label = getImageLabel(imageName);
+        ImageStream currentImageStream = client.imageStreams().withName(buildConfigName).get();
+        if (currentImageStream == null) {
+            throw new MojoExecutionException("Could not find a current ImageStream with name " + buildConfigName + " in namespace " + namespace);
+        }
+        String tagSha = findTagSha(currentImageStream);
+        String name = buildConfigName + "@" + tagSha;
+        String kind = "ImageStreamImage";
+
+        ImageStreamSpec spec = is.getSpec();
+        if (spec != null) {
+            spec = new ImageStreamSpec();
+            is.setSpec(spec);
+        }
+        List<TagReference> tags = spec.getTags();
+        if (tags == null) {
+            tags = new ArrayList<>();
+            spec.setTags(tags);
+        }
+        TagReference tag = null;
+        if (tags.isEmpty()) {
+            tag = new TagReferenceBuilder().build();
+            tags.add(tag);
+        } else {
+            tag = tags.get(tags.size() - 1);
+        }
+        ObjectReference from = tag.getFrom();
+        if (from == null) {
+            from = new ObjectReference();
+            tag.setFrom(from);
+        }
+
+        boolean answer = false;
+        if (!Objects.equals(label, tag.getName())) {
+            tag.setName(label);
+            answer = true;
+        }
+        if (!Objects.equals(kind, from.getKind())) {
+            from.setKind(kind);
+            answer = true;
+        }
+        if (!Objects.equals(namespace, from.getNamespace())) {
+            from.setNamespace(namespace);
+            answer = true;
+        }
+        if (!Objects.equals(name, from.getName())) {
+            from.setName(name);
+            answer = true;
+        }
+        if (answer) {
+            log.info("Updated ImageStream " + buildConfigName + " to namespace: " + namespace + " name: " + name);
+        }
+        return answer;
+    }
```

Comment:
the purpose of the updated ImageStream is not for the current build namespace; its to be able to install this image in a different namespace/cluster; so there's no need to update the current namespace at all - rather its a way of getting a reference to where the image stream is so we can tag it into different namespaces